src/libcollections/vec.rs

```diff
@@ -206,7 +209,7 @@ impl<T> Vec<T> {
                                .expect("capacity overflow");
             let ptr = unsafe { allocate(size, mem::min_align_of::<T>()) };
             if ptr.is_null() { ::alloc::oom() }
-            Vec { ptr: unsafe { NonZero::new(ptr as *mut T) }, len: 0, cap: capacity }
+            Vec { ptr: unsafe { NonZero::new(ptr as *mut T).unwrap() }, len: 0, cap: capacity }
```

Comment:
This (and a few other places) should match on the result of `NonZero::new` instead of manually checking for NULL after allocating - the result should be identical even unoptimized.